packages/tools-language/test/properties.test.ts

```diff
@@ -90,4 +95,61 @@ describe("Language > Props", () => {
       )
     ).toEqual({ x: 123, y: "test" });
   });
+
+  type BaseType = { base: string };
+  type ExtendedType = BaseType & { extended?: number };
+
+  const baseObj: BaseType = { base: "base object" };
+  const extendedProps: Omit<ExtendedType, keyof BaseType> = { extended: 12345 };
+
+  test("extendObject adds the extended props", () => {
+    const copy = { ...baseObj };
+    expect(extendObject<BaseType, ExtendedType>(copy, extendedProps)).toEqual({
+      ...baseObj,
+      ...extendedProps,
+    });
+  });
+
+  test("extendObject returns the original object reference", () => {
+    const copy = { ...baseObj };
+    expect(extendObject<BaseType, ExtendedType>(copy, extendedProps)).toBe(
+      copy
+    );
+  });
+
+  test("extendObject makes no changes when extended props are optional and are not given", () => {
+    const copy = { ...baseObj };
+    expect(extendObject<BaseType, ExtendedType>(copy, {})).toEqual(baseObj);
+  });
+
+  test("extendObjectArray adds the extended props to each object in the array", () => {
+    const copy = [{ ...baseObj }];
+    const extended = extendObjectArray<BaseType, ExtendedType>(
+      copy,
+      extendedProps
+    );
+    expect(extended).toBeArrayOfSize(1);
+    expect(extended).toIncludeSameMembers([
```

Comment:
Isn't `toEqual` sufficient here?
```suggestion
    expect(extended).toEqual([
```